tests/unit/services/s3/test_virtual_host.py

```diff
@@ -21,6 +21,9 @@ def request(self, request: Request, server: str | None = None) -> Response:
         self.requests.put((request, server))
         return Response()
 
+    def close(self):
+        pass
+
```

Comment:
I believe this has been superseded by #10099, but sadly I didn't put it at the same spot 😬 we might need to rebase before merging. Sorry!